ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -143,32 +154,32 @@
       case PutOperation:
         eligibleReplicas =
             getEligibleReplicas(partitionId, datacenterName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER));
-        successTarget = routerConfig.routerGetEligibleReplicasByStateEnabled ? Math.max(eligibleReplicas.size() - 1,
+        diskSuccessTarget = routerConfig.routerGetEligibleReplicasByStateEnabled ? Math.max(eligibleReplicas.size() - 1,
             routerConfig.routerPutSuccessTarget) : routerConfig.routerPutSuccessTarget;
-        parallelism = routerConfig.routerGetEligibleReplicasByStateEnabled ? eligibleReplicas.size()
+        diskParallelism = routerConfig.routerGetEligibleReplicasByStateEnabled ? eligibleReplicas.size()
             : routerConfig.routerPutRequestParallelism;
+        crossColoEnabled = false;
         break;
       case DeleteOperation:
-        successTarget = routerConfig.routerDeleteSuccessTarget;
-        parallelism = routerConfig.routerDeleteRequestParallelism;
+        diskSuccessTarget = routerConfig.routerDeleteSuccessTarget;
+        diskParallelism = routerConfig.routerDeleteRequestParallelism;
         crossColoEnabled = true;
         eligibleReplicas = getEligibleReplicas(partitionId, null,
             EnumSet.of(ReplicaState.BOOTSTRAP, ReplicaState.STANDBY, ReplicaState.LEADER));
         break;
       case TtlUpdateOperation:
-        successTarget = routerConfig.routerTtlUpdateSuccessTarget;
-        parallelism = routerConfig.routerTtlUpdateRequestParallelism;
+        diskSuccessTarget = routerConfig.routerTtlUpdateSuccessTarget;
+        diskParallelism = routerConfig.routerTtlUpdateRequestParallelism;
         crossColoEnabled = true;
         eligibleReplicas = getEligibleReplicas(partitionId, null,
             EnumSet.of(ReplicaState.BOOTSTRAP, ReplicaState.STANDBY, ReplicaState.LEADER));
         break;
       default:
         throw new IllegalArgumentException("Unsupported operation: " + routerOperation);
     }
-    if (parallelism < 1) {
-      throw new IllegalArgumentException("Parallelism has to be > 0. Configured to be " + parallelism);
+    if (diskParallelism < 1) {
```

Comment:
I think this check also applies to cloudParallelism.
Also, could you add additional info in the exception message to record operation type? (so far, only PUT operation has dynamic parallelism but it's still good to add type into error message)